prov/gni/src/gnix_msg.c

```diff
@@ -1888,6 +1894,48 @@ static int __smsg_eager_msg_w_data(void *data, void *msg)
 	req = _gnix_match_tag(posted_queue, hdr->msg_tag, 0, FI_PEEK, NULL,
 			      &vc->peer_addr);
 	if (req) {
+		if (req->type == GNIX_FAB_RQ_MRECV) {
+
+			mrecv_req = req;
+			req = _gnix_fr_alloc(ep);
```

Comment:
It's confusing to me why you assign mrecv_req to req, rather than just allocate new fab request.  Is there less stuff you have to copy if you do it in that direction?  If not, I think it would be easier to follow if it was the other way around.